dist/ethereum.js

```diff
@@ -2580,23 +2583,31 @@ var requestManager = function() {
     var send = function (data, callback) {
         /*jshint maxcomplexity: 8 */
 
-        // FORMAT BASED ON ONE FORMATTER function
-        if(typeof data.inputFormatter === 'function') {
-            data.params = Array.prototype.map.call(data.params, function(item, index){
-                // format everything besides the defaultblock, which is already formated
-                return (!data.addDefaultblock || index+1 < data.addDefaultblock) ? data.inputFormatter(item) : item;
-            });
+        if(!utils.isArray(data))
+            data = [data];
 
-        // FORMAT BASED ON the input FORMATTER ARRAY
-        } else if(data.inputFormatter instanceof Array) {
-            data.params = Array.prototype.map.call(data.inputFormatter, function(formatter, index){
-                // format everything besides the defaultblock, which is already formated
-                return (!data.addDefaultblock || index+1 < data.addDefaultblock) ? formatter(data.params[index]) : data.params[index];
-            });
-        }
+        var payload = [];
 
 
-        var payload = jsonrpc.toPayload(data.method, data.params);
+        data.forEach(function(item){
+            // FORMAT BASED ON ONE FORMATTER function
+            if(typeof item.inputFormatter === 'function') {
+                item.params = Array.prototype.map.call(item.params, function(param, index){
+                    // format everything besides the defaultblock, which is already formated
+                    return (!item.addDefaultblock || index+1 < item.addDefaultblock) ? item.inputFormatter(param) : param;
+                });
+
+            // FORMAT BASED ON the input FORMATTER ARRAY
+            } else if(item.inputFormatter instanceof Array) {
+                item.params = Array.prototype.map.call(item.inputFormatter, function(formatter, index){
+                    // format everything besides the defaultblock, which is already formated
+                    return (!item.addDefaultblock || index+1 < item.addDefaultblock) ? formatter(item.params[index]) : item.params[index];
+                });
+            }
+
+            payload.push(jsonrpc.toPayload(item.method, item.params));
```

Comment:
1. You should have used `map`, not `forEach`.
2. I don't like calling `typeof` and `instanceof` directly. We have utils methods `isArray`, `isFunction` to do that.
3. I don't like the fact, that we are checking if formatters are array or functions. You are overusing lack of types in javascript.
4. This function is called !send! Do you think that bringing formatters here is proper?